### ISSUE TITLE
uninformative error msg 

### ISSUE THREAD

**Issue by berndbischl** [schalkdaniel/compboost#239]
```
library(mlbench)
data(Sonar)

cboost = Compboost$new(data = Sonar, target = "Class", loss = BinomialLoss)
cboost$addBaselearner(feature = c("V1", "V2"), id = "foo", PolynomialBlearner)
cboost$train()
```

```
Error in new_CppObject_xp(fields$.module, fields$.pointer, ...) :
  Cannot convert object to an environment: [type=character; target=ENVSXP].
```

**Comment by schalkdaniel** [comment]
You are right, we should check if the user passes an initialized loss object (which is required here):
```
library(mlbench)
data(Sonar)

cboost = Compboost$new(data = Sonar, target = "Class", loss = BinomialLoss$new())
cboost$addBaselearner(feature = c("V1", "V2"), id = "foo", PolynomialBlearner)
cboost$train()
```

**Comment by berndbischl** [comment]
yes, that was the problem. took some time to see this
